utils/rss.ts

```diff
@@ -0,0 +1,63 @@
+import fs from 'node:fs';
+import path from 'node:path';
+import { Feed } from 'feed';
+import { author, site } from '../config/index.json';
+import { getContentList, IContent, sortByDate } from './content';
+
+const { siteName, siteTitle, siteUrl } = site;
+
+export const notesContent = getContentList('notes');
+export const articlesContent = getContentList('articles');
+
+export function generateRSS() {
+  const year = new Date().getFullYear();
+  try {
+    console.log('🔃 - Generating RSS feed at rss.xml');
+    const feed = new Feed({
+      title: siteName,
+      description: siteTitle,
+      id: siteUrl,
+      link: siteUrl,
+      favicon: `${siteUrl}/favicon.ico`,
+      feedLinks: {
+        rss: `${siteUrl}/rss.xml`,
+      },
+      image: `${siteUrl}/Logo.png`,
+      copyright: `${year} ${author.name}`,
+      language: 'en',
+      author: {
+        name: author.name,
+        link: author.twitterHandle,
+      },
+    });
+
+    articlesContent.sort(sortByDate).forEach((contentItem: IContent) => {
+      const { title, previewImage, date, slug, description } = contentItem;
```

Comment:
removed `id`, because it was declared but never used

Reply:
Anything unused - eg. a destructured property like this or an unused import - can be PRed to upstream in a separate new PR

Reply:
created a PR in upstream
- https://github.com/vickonrails/next-starter-peacock/pull/26